### ISSUE TITLE
Try label smoothing

### ISSUE THREAD

**Issue by wingman-jr-addon** [wingman-jr-addon/wingman_jr#63]
In the past, I've tried normal label smoothing with the common alpha=0.1, and I saw slightly worse results.
Because I use a weighted form of categorical cross entropy, in the past I've thought I probably start to reap a few of the benefits of label smoothing by penalizing incorrect guesses by taking a graded approach. However, I was thinking about it again yesterday and I think there's still advantage in using label smoothing, but perhaps I need to similarly introduce smoothing in a graded way. In some ways what I am doing is categorical, but in some ways it is also like imposing a coarse metric on the output, so it makes sense to introduce uncertainty across classes as e.g. the bell curve of uncertainty around a grade.

**Comment by wingman-jr-addon** [comment]
The results here were interesting.

The training definitely went fast initially. Teal is with, grey is without.
![image](https://user-images.githubusercontent.com/55339824/82149093-db5c0000-981b-11ea-942b-382824c16af4.png)

However, validation started to peter out much earlier than usually and it appeared to be overfitting sooner.
With label smoothing:
![image](https://user-images.githubusercontent.com/55339824/82149252-3c83d380-981c-11ea-94be-319b4ff57d4c.png)
Without label smoothing:
![image](https://user-images.githubusercontent.com/55339824/82149373-5ae9cf00-981c-11ea-9bf2-069378e44461.png)

This was with the following label smoothing values - index this array instead of using `to_categorical`:
```
    smoothed_categorical = [
            np.array([0.80, 0.12, 0.05, 0.03]),
            np.array([0.08, 0.80, 0.08, 0.04]),
            np.array([0.04, 0.08, 0.80, 0.08]),
            np.array([0.03, 0.12, 0.05, 0.80])
        ]
```
While the idea may still have some promise, I think there are other avenues to explore first.
